### DISCUSSION TITLE
Is it possible to use remote images to set as virtual background?

### DISCUSSION THREAD

**Discussion by tremzo** [ant-media/Ant-Media-Server#5680] [FAQs] [answered]
Is it possible to fetch images from a remote location to set as the background image? Currently, if I try to set the virtual image using a URL it doesn't work. Only if I have the images in a local directory does the virtual background get set.

**Comment by zerostack0** [comment, marked as answer]
It's normal because when you set an image URL, it tries to fetch it from the internet in every frame. Instead of using URL directly, you can save image in base64 format and use it.

```
const getBase64FromUrl = async (url) => {
            const data = await fetch(url);
            const blob = await data.blob();
            return new Promise((resolve) => {
                const reader = new FileReader();
                reader.readAsDataURL(blob);
                reader.onloadend = () => {
                    const base64data = reader.result;
                    resolve(base64data);
                }
            });
        }
```

```
    function setImageBase64(url) {
        let imageBase64 = document.getElementById("imageBase64"); // be sure that an image element with imageBase64 id is exist
        getBase64FromUrl(url).then((result) => {
            imageBase64.src = result;
        });
    }
```

`    setImageBase64("https://marketplace.canva.com/EAD2962NKnQ/2/0/1600w/canva-rainbow-gradient-pink-and-purple-virtual-background-_Tcjok-d9b4.jpg");`

The result :

![Screenshot 2023-10-12 at 11 10 30](https://github.com/ant-media/Ant-Media-Server/assets/23143596/2cc015aa-cd3e-4126-9795-090a25daee01)
